variables.tf

```diff
@@ -363,3 +363,78 @@ variable "create_service_link_role" {
   type        = bool
   default     = true
 }
+
+
+# Auto-tune
+variable "auto_tune_options" {
+  description = "Options for auto-tune configuration"
+  type        = any
+  default = {
+    # desired_state = "ENABLED"
+    # desired_state = "DISABLED"
+    # rollback_on_disable = "NO_ROLLBACK"
+    # rollback_on_disable = "DEFAULT_ROLLBACK"
+    # maintenance_schedule = {
+    #   start_at = "2000-01-01T00:00:00.00Z"
+    #   cron_expression_for_recurrence = "0 0 * * FRI"
+    #   duration = {
+    #     value = 10
+    #     unit = "HOURS"  
+    #   }
+    # }
+  }
+}
+
+variable "auto_tune_options_desired_state" {
+  description = "Whether auto-tune is enabled or not"
+  type        = string
+  default     = "ENABLED"
+  validation {
+    condition     = can(regex("^(ENABLED|DISABLED)$", var.auto_tune_options_desired_state))
+    error_message = "The value of auto_tune_options_desired_state must be one of: ENABLED, DISABLED."
+  }
+}
+
+variable "auto_tune_options_rollback_on_disable" {
+  description = "Behaviour if auto-tune is disabled"
+  type        = string
+  default     = "NO_ROLLBACK"
+  validation {
+    condition     = can(regex("^(NO_ROLLBACK|DEFAULT_ROLLBACK)$", var.auto_tune_options_rollback_on_disable))
+    error_message = "The value of auto_tune_options_rollback_on_disable must be one of: NO_ROLLBACK, DEFAULT_ROLLBACK."
+  }
+}
+
+variable "auto_tune_options_start_at" {
+  description = "Timestamp in which the maintenance window starts"
+  type        = string
+  # default = "2000-01-01T00:00:00.00Z"
+  default = null
+  validation {
+    condition     = can(regex("^[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{2})?(Z|([-+]{1}[0-9]{2}:[0-9]{2))$", var.auto_tune_options_rollback_on_disable))
```

Comment:
@rogerfmglobal here `var.auto_tune_options_rollback_on_disable` should be `var.auto_tune_options_start_at`, otherwise you'll get this error:

```
╷
│ Error: Invalid variable validation condition
│
│   on .terraform/modules/aws_es/variables.tf line 414, in variable "auto_tune_options_start_at":
│  414:     condition     = can(regex("^[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{2})?(Z|([-+]{1}[0-9]{2}:[0-9]{2))$", var.auto_tune_options_rollback_on_disable))
│
│ The condition for variable "auto_tune_options_start_at" must refer to var.auto_tune_options_start_at in order to test incoming values.
╵
```
Can you change it? Thanks in advance!

Reply:
Another thing, if it's null the condition fails too:

```
$ terraform plan
╷
│ Error: Invalid value for variable
│
│   on .terraform/modules/aws_es/variables.tf line 408:
│  408: variable "auto_tune_options_start_at" {
│     ├────────────────
│     │ var.auto_tune_options_start_at is null
│
│ The value of auto_tune_options_start_at must be a valid RFC-3339 timestamp.
│
│ This was checked by the validation rule at .terraform/modules/aws_es/variables.tf:413,3-13.
╵
```

Reply:
About that error, https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/elasticsearch_domain#auto_tune_options says:
[maintenance_schedule] - (Required if rollback_on_disable is set to DEFAULT_ROLLBACK

I guess that is the case in the error you posted.

I'm not sure how to handle this or if the module even should do it. I mean: should the module check the value for rollback_on_disable to set a default value for start_at that equals null or a timestamp? I'll take a look.

Reply:
Fixed validation and added condition:

condition     = var.auto_tune_options_start_at == null || can(regex("^[0-9]{4}\\-[0-9]{2}\\-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}(\\.[0-9]{2})?(Z|([-+]{1}[0-9]{2}:[0-9]{2))$", var.auto_tune_options_start_at))